### PR TITLE
[Enhancement] Improve orc/text code

### PR DESCRIPTION
**Pull request by Smith-Cruise**
## Why I'm doing:
1. csv reader will not release memory util the entire file read is finished. (Caused by SharedBuffer)
2. orc reader should ignore skip stripe caused by scan range mismatch.
3. add debug function in SharedBuffer, easier to print log

## What I'm doing:
just do it.

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [x] 3.1
  - [x] 3.0
  - [ ] 2.5

### CODE REVIEW

**Review by dirtysalt — APPROVED** [review]

**Review by letian-jiang — APPROVED** [review]

File: be/src/io/shared_buffered_input_stream.cpp [hunk 1]
```diff
@@ -36,6 +38,12 @@ void SharedBufferedInputStream::SharedBuffer::align(int64_t align_size, int64_t
     }
 }
 
+std::string SharedBufferedInputStream::SharedBuffer::debug() const {
```
Comment (dirtysalt) [inline be/src/io/shared_buffered_input_stream.cpp:41]: debug_string

**Review by satanson — APPROVED** [review]

File: be/src/exec/hdfs_scanner_text.cpp [hunk 2]
```diff
@@ -443,8 +455,9 @@ Status HdfsTextScanner::_create_or_reinit_reader() {
 Status HdfsTextScanner::_setup_io_ranges() const {
     if (_shared_buffered_input_stream != nullptr) {
         std::vector<io::SharedBufferedInputStream::IORange> ranges{};
-        for (int64_t offset = 0; offset < _scanner_params.file_size;) {
-            const int64_t remain_length = std::min(config::text_io_range_size, _scanner_params.file_size - offset);
+        const int64_t scan_range_end = _scanner_params.scan_range->offset + _scanner_params.scan_range->length;
```
Comment (dirtysalt) [inline be/src/exec/hdfs_scanner_text.cpp:458]: do we need to get min(file_size,scan_end) ?

File: be/src/formats/orc/apache-orc/c++/src/Reader.cc [hunk 3]
```diff
@@ -1152,7 +1154,8 @@ void RowReaderImpl::startNextStripe() {
             currentStripe += 1;
             currentRowInStripe = 0;
 
-            if (contents->stream->get_lazy_column_coalesce_counter() != nullptr) {
+            // We do not count the skipped stripes because the scan range does not match
+            if (!skipStripeByScanRangeMisMatch && contents->stream->get_lazy_column_coalesce_counter() != nullptr) {
```
Comment (dirtysalt) [inline be/src/formats/orc/apache-orc/c++/src/Reader.cc:1158]: why not use skip strip?

File: be/src/exec/hdfs_scanner_text.cpp (hunk 2 above)
Comment (Smith-Cruise) [reply be/src/exec/hdfs_scanner_text.cpp:458]: scan_range_end must <= file_size

File: be/src/io/shared_buffered_input_stream.cpp (hunk 1 above)
Comment (Smith-Cruise) [reply be/src/io/shared_buffered_input_stream.cpp:41]: change next time, some people is waiting for this pr

**Review by wyb — APPROVED** [review]

File: be/src/exec/hdfs_scanner_text.cpp (hunk 2 above)
Comment (sduzh) [reply be/src/exec/hdfs_scanner_text.cpp:458]: > scan_range_end must <= file_size

This is not a justifiable rationale. I'm afraid urgency alone does not justify compromising on quality assurance.
Besides, resolving inconsistent naming is quite simple,  it should only require a small time investment.

File: be/src/io/shared_buffered_input_stream.cpp (hunk 1 above)
Comment (sduzh) [reply be/src/io/shared_buffered_input_stream.cpp:41]: > change next time, some people is waiting for this pr

This is not a justifiable rationale
I'm afraid urgency alone does not justify compromising on quality assurance.
Besides, it's merely a name change, it should only require a small time investment.

Comment (Smith-Cruise) [reply be/src/io/shared_buffered_input_stream.cpp:41]: I will submit another pr to rename it.

**Pull request merged**